generator/src/googleapis/codegen/api.py

```diff
@@ -331,8 +331,11 @@ def _NormalizeUrlComponents(self):
     scheme = url_parts.scheme or 'https'
     service_host = url_parts.netloc or _DEFAULT_SERVICE_HOST
     base_path = url_parts.path
-    if not root_url:
+    if root_url:
+      self._api.SetTemplateValue('rootUrlTemplate', root_url.replace('googleapis.com', 'UNIVERSE_DOMAIN'))
```

Comment:
This might break though, if users have a rooturl like:
`usergoogleapis.com`, it would be changed to `userUNIVERSE_DOMAIN`.

Reply:
I am not sure though how big of a usecase this is... a client using TPC and their domain name has `googleapis.com`

Reply:
we could replace `.googleapis.com` with `.UNIVERSE_DOMAIN` if you think that would be safer.